modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -3686,21 +3694,32 @@ else if (valueString.isEmpty()) {
 		}
 		else if (sqlType == Types.DECIMAL) {
 			if (Validator.isNull(String.valueOf(value))) {
-				value = BigDecimal.ZERO;
+				preparedStatement.setBigDecimal(index, null);
```

Comment:
Couldn't we use preparedStatement.setNull here?

Reply:
Yes! I just updated the PR